File: packages/python-sdk/src/blindfold/regex/detectors/es/dni.py
"""Spanish DNI (Documento Nacional de Identidad) detector"""

import re

from ...base import RegexDetector
from ...entities import EntityType
from ...registry import register_region
from ...validators import es_dni_letter


@register_region("es")
class DniDetector(RegexDetector):
    entity_type = EntityType.ES_DNI
    score = 0.90
    context_keywords = [
        "dni", "documento nacional", "identidad",
    ]
    context_required = False
    context_window = 50

    # 8 digits + letter
    pattern = re.compile(
        r"\b\d{8}[-\s]?[A-Z]\b",
        re.IGNORECASE,
    )
    validator = staticmethod(es_dni_letter)

File: packages/js-sdk/src/regex/detectors/es/dni.ts
// Spanish DNI (Documento Nacional de Identidad) detector

import { RegexDetector } from '../../base'
import { EntityType } from '../../entities'
import { registerRegion } from '../../registry'
import { esDniLetter } from '../../validators'

class EsDniDetector extends RegexDetector {
  entityType = EntityType.ES_DNI
  score = 0.9
  contextKeywords = ['dni', 'documento nacional', 'identidad']
  contextRequired = false

  pattern = /\b\d{8}[-\s]?[A-Z]\b/gi
  validator = esDniLetter
}

registerRegion('es', EsDniDetector)

export { EsDniDetector }
